modules/Hoot/ui/modals/modifyFolder.js

```diff
@@ -12,40 +12,55 @@ import { modifyDatasetForm } from '../../config/domMetadata';
 
 export default class ModifyFolder {
     constructor( d ) {
-        this.data       = d.data;
+        this.data = d;
 
         //get list of folder ids and all their descendents
         function getDescendents(ids, folders) {
-            let children = folders.filter(f => ids.includes(f.parentId))
-                              .map(f => f.id);
+
+            let children = folders.filter(f => ids.findIndex( getId => getId.id === f.parentId) !== -1 );
+
             if (children.length) {
                 return [...new Set(ids.concat(children).concat(getDescendents(children, folders)))];
             } else {
                 return ids;
             }
         }
-        let descendents = getDescendents([d.data.id], Hoot.folders._folders);
+        let descendents = getDescendents( d, Hoot.folders._folders);
 
         //filter out the folder itself
         //and all of it's descendents
         this.folderList = Hoot.folders._folders.filter(f => {
             return !descendents.includes(f.id);
         });
+
         this.form       = modifyDatasetForm.call( this );
     }
 
     render() {
         // remove layer name input
         this.form.splice( 2, 1 );
+
         this.pathName = _get( _find( this.folderList, folder => folder.id === this.data.parentId ), 'path' ) || '/';
 
         // Because dataset and folder share the same settings we had to set a trigger here to tell the formFactory
         // that we want root in the path dropdown
         const pathComboboxSettings = _find( this.form, formItem => formItem.itemKey === 'path' );
         pathComboboxSettings.includeRoot = true;
 
+        if ( this.data.length > 1 ) {
+
+            this.form.splice( 0, 1 );
+
+            let that = this;
+
+            that = that.folderList.filter( f => this.data.findIndex( folder => folder.parentId === f.id || folder.id === f.id || folder.name === f.name ) === -1 );
+
+            this.form[0].data = that;
+
+        }
```

Comment:
what's this doing?

Reply:
Moving multiple folders ,and individual folders for that matter, we shouldn't have the option to move a folder and its children into the selected folder's parent.

We also shouldn't be capable of moving a selected folder or folders into a child of that folder. 

This method is attempting to present the user with a meaningful list of options for moving the selected folder.